cli/core/utils.go

```diff
@@ -337,10 +345,12 @@ func PrepareAccount(owner *string, chainID *big.Int, noSend bool) (*Owner, error
 			return nil, err
 		}
 
-		auth.GasPrice = nil             // big.NewInt(10)  // Gas price to use for the transaction execution (nil = gas price oracle)
-		auth.GasFeeCap = big.NewInt(10) // big.NewInt(10) // Gas fee cap to use for the 1559 transaction execution (nil = gas price oracle)
-		auth.GasTipCap = big.NewInt(2)  // big.NewInt(2) // Gas priority fee cap to use for the 1559 transaction execution (nil = gas price oracle)
-		auth.GasLimit = 21000
+		if !isSimulatingGas {
+			auth.GasPrice = nil             // big.NewInt(10)  // Gas price to use for the transaction execution (nil = gas price oracle)
+			auth.GasFeeCap = big.NewInt(10) // big.NewInt(10) // Gas fee cap to use for the 1559 transaction execution (nil = gas price oracle)
+			auth.GasTipCap = big.NewInt(2)  // big.NewInt(2) // Gas priority fee cap to use for the 1559 transaction execution (nil = gas price oracle)
+			auth.GasLimit = 21000
```

Comment:
specifying GasLimit manually leads to the txn not being simulated (which is what we want in this branch)